solutions/cfgym-102951C.mdx

```diff
@@ -74,4 +76,57 @@ int main() {
 
 </CPPSection>
 
+<JavaSection>
+	
+``` java
+import java.io.*;
+
+import java.util.*;
+
+
+public class Main {
+	// CodeSnip{Kattio}
+	public static void main(String[] args) {
+		Kattio io = new Kattio();
+
+		int n = io.nextInt();		
+		int[] a = new int[n + 1], b = new int[n + 1], c = new int[n + 1], lis = new int[n + 1];
+		int[] pos = new int[n + 2];
+
+		for(int i = 1; i <= n; i++)
+		{
+			a[i] = io.nextInt();
+			 //Pos is the inverse of a
+			pos[a[i]] = i;
+		}
+
+		for(int i = 1; i <= n; i++)
+		{
+			b[i] = io.nextInt();
+		}
+
+		for(int i = 1; i <= n; i++)
+		{
+			c[i] = pos[b[i]];
+		}
+
+		// Length of the LIS
+		int len = 0; 
+		for(int i = 1; i <= n; i++)
+		{
+			// Finding LowerBound
```

Comment:
something more descriptive than "finding lowerbound" would help

Reply:
Hi, I've updated the comment and committed it. Lemme know if it can be improved even further.